scripts/lib/CIME/namelist.py

```diff
@@ -171,6 +171,20 @@ def is_valid_fortran_name(string):
     """
     return FORTRAN_NAME_REGEX.search(string) is not None
 
+def fortran_name_index(string):
```

Comment:
If a valid index exists return that value.   Note that we are not yet handling array sections such as (:2) or (3:5)